core/loadpoint.go

```diff
@@ -879,14 +882,18 @@ func (lp *LoadPoint) effectiveCurrent() float64 {
 
 // elapsePVTimer puts the pv enable/disable timer into elapsed state
 func (lp *LoadPoint) elapsePVTimer() {
-	lp.pvTimer = lp.clock.Now().Add(-lp.Disable.Delay)
-	lp.guardUpdated = lp.clock.Now().Add(-lp.GuardDuration)
+	lp.log.DEBUG.Printf("pv timer elapse")
+
+	lp.pvTimer = elapsed
+	lp.guardUpdated = elapsed
+
 	lp.publishTimer(pvTimer, 0, timerInactive)
 }
 
 // resetPVTimerIfRunning resets the pv enable/disable timer to disabled state
 func (lp *LoadPoint) resetPVTimerIfRunning(typ ...string) {
-	if lp.pvTimer.IsZero() {
+	// https://github.com/evcc-io/evcc/issues/2289
+	if lp.pvTimer.IsZero() || lp.pvTimer.Equal(elapsed) {
```

Comment:
Für unser Problem würden wir hier ausnutzen, dass die Funktion `scalePhase()` bereits `elapsePVTimer()` aufruft, und wir in diesem Fall nichts mehr unternehmen müssen. Top!

Allerdings könnte man so mit der Funktion einen elapsed Timer nicht mehr resetten (Falls man das mal möchte ;))
Was ich so erstmal nicht von der Funktion erwarten würde.

Was hältst du davon, dass stattdessen direkt im `evChargeStopHandler` machen? In etwa so:

```go
func (lp *LoadPoint) evChargeStopHandler() {
	lp.log.INFO.Println("stop charging <-")
	lp.pushEvent(evChargeStop)

	// soc update reset
	lp.socUpdated = time.Time{}

	// reset pv enable/disable timer
    if !lp.pvTimer.Equal(elapsed) {
	    lp.resetPVTimerIfRunning()
    }
}
```